### PR TITLE
add contextlib.chdir

### PR DESCRIPTION
**Pull request by graingert**
see https://github.com/python/cpython/pull/28271

Is it too early to start adding 3.11 types to typeshed? I'm looking to add it here so I can backport the types cleanly into contextlib2

### CODE REVIEW

File: stdlib/contextlib.pyi [hunk 1]
```diff
@@ -124,3 +124,12 @@ elif sys.version_info >= (3, 7):
         def __init__(self: nullcontext[_T], enter_result: _T) -> None: ...
         def __enter__(self) -> _T: ...
         def __exit__(self, *exctype: Any) -> None: ...
+
+if sys.version_info >= (3, 11):
+    _T_fd_or_any_path = TypeVar("_T_fd_or_any_path", bound=int | StrOrBytesPath)
+    class chdir(AbstractContextManager[_T_fd_or_any_path]):
+        path: _T_fd_or_any_path
+        _old_cwd: list[str]
```
[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stdlib/contextlib.pyi:133]: @graingert could you respond here?

[the remark this answers is not in the document]
Comment (graingert) [reply stdlib/contextlib.pyi:133]: Doesn't seem so bad to have it? I don't mind either way

[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stdlib/contextlib.pyi:133]: I'd prefer we keep it out until and unless we get people who actually need it asking for it.

File: stdlib/contextlib.pyi [hunk 2]
```diff
@@ -124,3 +125,12 @@ elif sys.version_info >= (3, 7):
         def __init__(self: nullcontext[_T], enter_result: _T) -> None: ...
         def __enter__(self) -> _T: ...
         def __exit__(self, *exctype: Any) -> None: ...
+
+if sys.version_info >= (3, 11):
+    _T_fd_or_any_path = TypeVar("_T_fd_or_any_path", bound=int | StrOrBytesPath)
+    class chdir(AbstractContextManager[None], Generic[_T_fd_or_any_path]):
+        path: _T_fd_or_any_path
+        _old_cwd: list[str]
```
Comment (graingert) [inline stdlib/contextlib.pyi:133]:
```suggestion
```

**Review by JelleZijlstra — APPROVED** [review]

**Review by Akuli — APPROVED** [review]

**Pull request merged**